enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -219,6 +204,94 @@ func (l *FileCacheLoader) GetSnapshot(ctx context.Context, key *fcpb.SnapshotKey
 }
 
+func (l *FileCacheLoader) actionResultToManifest(snapshotActionResult *repb.ActionResult) (*fcpb.SnapshotManifest, error) {
+	snapMetadata := snapshotActionResult.GetExecutionMetadata().GetAuxiliaryMetadata()
+	if len(snapMetadata) != 1 {
+		return nil, status.InternalErrorf("expected vm config in snapshot auxiliary metadata")
+	}
+
+	vmConfig := &fcpb.VMConfiguration{}
+	if err := snapMetadata[0].UnmarshalTo(vmConfig); err != nil {
+		return nil, status.WrapErrorf(err, "unmarshall vm config")
+	}
+
+	manifest := &fcpb.SnapshotManifest{
+		VmConfiguration: vmConfig,
+		Files:           []*repb.FileNode{},
+		ChunkedFiles:    []*fcpb.ChunkedFile{},
+	}
+
+	for _, fileMetadata := range snapshotActionResult.OutputFiles {
+		manifest.Files = append(manifest.Files, &repb.FileNode{
+			Name:   fileMetadata.GetPath(),
+			Digest: fileMetadata.GetDigest(),
+		})
+	}
+
+	for _, chunkedFileMetadata := range snapshotActionResult.OutputDirectories {
+		tree, err := l.chunkedFileTree(chunkedFileMetadata)
+		if err != nil {
+			return nil, err
+		}
+
+		fileName := chunkedFileName(chunkedFileMetadata)
+		fileSize, err := chunkedFileProperty(tree, "size")
+		if err != nil {
+			return nil, err
+		}
+		chunkSize, err := chunkedFileProperty(tree, "chunk_size")
+		if err != nil {
+			return nil, err
+		}
+
+		chunks := make([]*fcpb.Chunk, 0, len(tree.GetRoot().GetFiles()))
+		for _, chunk := range tree.GetRoot().GetFiles() {
+			offset, err := strconv.ParseInt(chunk.GetName(), 10, 64)
+			if err != nil {
+				return nil, status.InternalErrorf("parse chunk offset: %s", chunk.GetName())
+			}
+			chunks = append(chunks, &fcpb.Chunk{
+				Offset: offset,
+				Digest: chunk.GetDigest(),
+			})
+		}
+
+		manifest.ChunkedFiles = append(manifest.ChunkedFiles, &fcpb.ChunkedFile{
+			Name:      fileName,
+			Size:      fileSize,
+			ChunkSize: chunkSize,
+			Chunks:    chunks,
+		})
+	}
+	return manifest, nil
+}
+
+func chunkedFileName(chunkedFileMetadata *repb.OutputDirectory) string {
+	return chunkedFileMetadata.Path
+}
+
+func chunkedFileProperty(chunkedFileTree *repb.Tree, propertyName string) (int64, error) {
+	for _, prop := range chunkedFileTree.GetRoot().GetNodeProperties().GetProperties() {
+		if prop.GetName() == propertyName {
+			return strconv.ParseInt(prop.GetValue(), 10, 64)
+		}
+	}
+	return 0, status.InternalErrorf("chunked file metadata missing %s property", propertyName)
+}
+
+func (l *FileCacheLoader) chunkedFileTree(chunkedFileMetadata *repb.OutputDirectory) (*repb.Tree, error) {
+	node := &repb.FileNode{Digest: chunkedFileMetadata.GetTreeDigest()}
+	buf, err := filecacheutil.Read(l.env.GetFileCache(), node)
+	if err != nil {
+		return nil, status.UnavailableErrorf("failed to read chunked file tree: %s", status.Message(err))
+	}
+	tree := &repb.Tree{}
+	if err := proto.Unmarshal(buf, tree); err != nil {
+		return nil, status.WrapError(err, "unmarshall chunked file tree")
+	}
+	return tree, nil
+}
+
```

Comment:
I think it might be good to have this code eventually if we want to share some file-downloading code w/ `dirtools.go`, but thinking about it a bit more, since the snapshotting stuff is very specialized (e.g. we lazily download everything, we have this special "chunk_size" prop...) I wonder if realistically we'll get around to that any time soon.

wdyt about keeping this code in our back pocket until we have a more solid plan for consolidating this with `dirtools.go`, and maybe just serializing the manifest proto to a single `[]byte` field in the ActionResult for now (e.g. `stdout_raw`), just to keep things simple?